src/main/java/org/broadinstitute/hellbender/tools/htsgetreader/HtsgetResponse.java

```diff
@@ -26,17 +46,78 @@
         @JsonProperty("class")
         private HtsgetClass dataClass;
 
+        @JsonIgnore
+        private static CloseableHttpClient client;
+
         public URI getUri() {
             return this.uri;
         }
 
         public Map<String, String> getHeaders() {
-            return this.headers;
+            return Collections.unmodifiableMap(this.headers);
         }
 
         public HtsgetClass getDataClass() {
             return this.dataClass;
         }
+
+        private static synchronized CloseableHttpClient getHttpClient() {
```

Comment:
Since this is a pretty substantial block of now duplicated code, let deduplicate it.  I would recommend just moving this getHttpClient method to a new class, maybe `HttpUtils`, and maybe just making it a singleton since it's threadsafe and we could share it.

Reply:
Fixed.